src/components/ha-expansion-panel.ts

```diff
@@ -0,0 +1,128 @@
+import {
+  css,
+  CSSResult,
+  customElement,
+  html,
+  LitElement,
+  property,
+  TemplateResult,
+} from "lit-element";
+import { fireEvent } from "../common/dom/fire_event";
+import "./ha-svg-icon";
+import { mdiChevronDown } from "@mdi/js";
+import { classMap } from "lit-html/directives/class-map";
+
+@customElement("ha-expansion-panel")
+class HaExpansionPanel extends LitElement {
+  @property({ type: Boolean, reflect: true }) expanded = false;
+
+  @property({ type: Boolean, reflect: true }) outlined = false;
+
+  protected render(): TemplateResult {
+    return html`
+      <div class="summary" @click=${this._toggleContainer}>
+        <slot name="title"></slot>
+        <ha-svg-icon
+          .path=${mdiChevronDown}
+          class="summary-icon ${classMap({ expanded: this.expanded })}"
+        ></ha-svg-icon>
+      </div>
+      <div
+        class="container ${classMap({ expanded: this.expanded })}"
+        @transitionend=${this._handleTransitionEnd}
+      >
+        <slot></slot>
+      </div>
+    `;
+  }
+
+  private _handleTransitionEnd() {
+    const container = this.shadowRoot!.querySelector(
+      ".container"
+    )! as HTMLDivElement;
+
+    if (this.expanded) {
+      container.style.height = "auto";
```

Comment:
You can just remove the inline styles, as the normal css rules will take over when it isn't animating.

Reply:
I need this because I'm setting the height explicit before I start the collapse animation

Reply:
I understand, but you can just remove it? As the css has `height: auto`?

Reply:
I'm setting explicit height in `_toggleContainer`, so the auto is overridden. This is used to remove explicit height when the expand ends. You suggested using `height: auto` for the expanded container.
I need an explicit height to have a transition.

Reply:
@bramkragten are you ok with this? I need to have a fixed height to show transitions. That's why I'm setting it before I start both transitions (expand and collapse). I'm also setting auto height when the expand transition ends.